charms/worker/k8s/lib/charms/k8s/v0/k8sd_api_manager.py

```diff
@@ -124,24 +126,22 @@ def check_status_code(cls, v):
             raise ValueError(f"Status code must be 200. Received {v}")
         return v
 
-    @validator("error_code", always=True)
-    def check_error_code(cls, v, values):
+    @model_validator(mode="after")
+    def check_error_code(self) -> Self:
```

Comment:
I would not expect `check_error_code` to return `self` but I guess this is for chaining checks? Is this a common pattern for charms?

Reply:
this is a pattern from pydantic validators
https://docs.pydantic.dev/2.8/concepts/validators/#model-validators